web/js/map-OpenLayers.js

```diff
@@ -115,6 +115,29 @@ var fixmystreet = fixmystreet || {};
         fixmystreet.markers.redraw();
       },
 
+      reorder_visible: function() {
+          var visible = ''
+          var $itemList = $('.item-list');
+          var $visibleReports = $('.visible-reports');
+          for (var i = 0; i < fixmystreet.markers.features.length; i++) {
+              var feature = fixmystreet.markers.features[i];
+              var $listItem = $itemList.find('#report-'+ feature.data.id);
+              if (feature.onScreen()) {
+                  visible += $listItem[0].outerHTML; // This is faster than using .append()
```

Comment:
This means each visible report is now in the DOM twice with the same ID. As long as nothing is relying on this (it's new here, so presumably not), I guess that's okay, but the worry is if something comes along and assumes it's unique in future.